op-program/client/interop/consolidate.go

```diff
@@ -122,13 +185,12 @@ func newConsolidateCheckDeps(chains []eth.ChainIDAndOutput, oracle l2.Oracle) (*
 	}
 
 	canonBlocks := make(map[eth.ChainID]*l2.CanonicalBlockHeaderOracle)
-	for _, chain := range chains {
-		output := oracle.OutputByRoot(common.Hash(chain.Output), chain.ChainID)
-		outputV0, ok := output.(*eth.OutputV0)
-		if !ok {
-			return nil, fmt.Errorf("unexpected output type: %T", output)
-		}
-		head := oracle.BlockByHash(outputV0.BlockHash, chain.ChainID)
+	for i, chain := range chains {
+		progress := transitionState.PendingProgress[i]
+		// This is the optimistic head. It's OK if it's replaced by a deposits-only block.
+		// Because by then the replacement block won't be used for hazard checks.
+		// TODO: for extra safety, ensure the l2 oracle used for checks isn't affected by block reexec.
```

Comment:
nit: Should have a ticket logged and referenced for this one too please.

Reply:
ticket logged and referenced. Good note.